contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -88,19 +95,138 @@ contract MixinSignatureValidator is
         );
     }
 
+    /// @dev Approves/unnapproves an OrderValidator contract to verify signatures on signer's behalf
+    ///      using the `OrderValidator` signature type.
+    /// @param validatorAddress Address of Validator contract.
+    /// @param approval Approval or disapproval of  Validator contract.
+    function setOrderValidatorApproval(
+        address validatorAddress,
+        bool approval
+    )
+        external
+        nonReentrant
+    {
+        address signerAddress = getCurrentContextAddress();
+        allowedOrderValidators[signerAddress][validatorAddress] = approval;
+        emit SignatureValidatorApproval(
+            signerAddress,
+            validatorAddress,
+            approval
+        );
+    }
+
+    /// @dev Verifies that a signature for an order is valid.
+    /// @param order The order.
+    /// @param signerAddress Address that should have signed the given order.
+    /// @param signature Proof that the order has been signed by signer.
+    /// @return True if the signature is valid for the given order and signer.
+    function isValidOrderSignature(
+        Order memory order,
+        address signerAddress,
+        bytes memory signature
+    )
+        public
```

Comment:
This is never actually used internally, right? We should make it `external` if not.